examples/solvers/chambolle_pock_deconvolve.py

```diff
@@ -0,0 +1,141 @@
+
+class Convolution(odl.Operator):
+    def __init__(self, space, kernel, adjkernel):
+        self.kernel = kernel
+        self.adjkernel = adjkernel
+        super().__init__(space, space, linear=True)
+
+    def _call(self, rhs, out):
+        scipy.ndimage.convolve(rhs, self.kernel, output=out.asarray(),
+                               mode='wrap')
+        return out
+
+    @property
+    def adjoint(self):
+        return Convolution(self.domain, self.adjkernel, self.kernel)
+
+
+def kernel(x):
+    mean = [0.1, 0.5]
+    std = [0.1, 0.1]
+    k = np.exp(-(((x[0] - mean[0]) / std[0]) ** 2 +
+                 ((x[1] - mean[1]) / std[1]) ** 2))
+    k = np.around(k, 8)
+    return k
+
+
+def adjkernel(x):
+    return kernel((-x[0], -x[1]))
+
+
+# Continuous definition of problem
+cont_space = odl.FunctionSpace(odl.Rectangle([-1, -1], [1, 1]))
+kernel_space = odl.FunctionSpace(cont_space.domain - cont_space.domain)
+
+# Discretization parameters
+n = 50
+npoints = np.array([n + 1, n + 1])
+npoints_kernel = np.array([2 * n + 1, 2 * n + 1])
+
+# Discretized spaces
+discr_space = odl.uniform_discr_fromspace(cont_space, npoints)
+discr_kernel_space = odl.uniform_discr_fromspace(kernel_space, npoints_kernel)
+
+# Discretize the functions
+disc_kernel = discr_kernel_space.element(kernel)
+disc_adjkernel = discr_kernel_space.element(adjkernel)
+
+# Initialize convolution operator
+conv = Convolution(discr_space, disc_kernel, disc_adjkernel)
+
+# Optional: Run diagnostics to assure the adjoint is properly implemented
+# odl.diagnostics.OperatorTest(conv_op).run_tests()
+
+# Initialize gradient operator
+grad = odl.Gradient(discr_space, method='forward')
+
+# Matrix of operators
+op = odl.ProductSpaceOperator([[conv], [grad]])
+
+# Load phantom
+discr_phantom = odl.util.phantom.shepp_logan(discr_space, modified=True)
+
+# Create data vector of convolved phantom
+data = conv(discr_phantom)
+
+# Starting point
+x = op.domain.one()
+
+# Operator norm, add 10 percent to ensure ||K||_2^2 * sigma * tau < 1
+op_norm = 1.1 * odl.operator.oputils.power_method_opnorm(op, 100)
+print('Norm of the product space operator: {}'.format(op_norm))
+
+# Create proximal operators for unconstrained primal variable
+proximal_primal = odl.solvers.proximal_zero(op.domain)
+
+# Create proximal operators for the dual variable
+
+# l2-data matching
+prox_convconj_l2 = odl.solvers.proximal_convexconjugate_l2(discr_space, g=data)
+
+# TV-regularization
+prox_convconj_l1 = odl.solvers.proximal_convexconjugate_l1(
+    grad.range, lam=0.01)
+
+# Combine proximal operators, order must correspond to the operator K
+proximal_dual = odl.solvers.combine_proximals(
+    [prox_convconj_l2, prox_convconj_l1])
+
+# Optionally pass partial to the solver to display intermediate results
+partial = (odl.solvers.util.PrintIterationPartial() &
+           odl.solvers.util.PrintTimingPartial() &
+           odl.solvers.util.ShowPartial())
```

Comment:
Off-topic: can't we come up with a better name than `Partial`? It doesn't say anything (at least to me).